packages/interop-tests/src/__test__/interop.test.ts

```diff
@@ -148,25 +192,79 @@ it('server + browser wallet interoperability test', async () => {
    */
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const postFundAPromise = fromEvent<SingleChannelOutput>(serverWallet as any, 'channelUpdated')
+  const postFundA = await fromEvent<SingleChannelOutput>(serverWallet as any, 'channelUpdated')
     .pipe(take(3))
     .toPromise();
 
+  await browserWallet.pushMessage(serverMessageToBrowserMessage(postFundA), 'dummyDomain');
+
+  await objectiveSuccededPromise;
+});
+
+it('browser wallet creates channel + cooperates with server wallet to fund channel', async () => {
+  const createChannelParams: CreateChannelParams = {
+    participants: [
+      {
+        participantId: browserAddress,
+        signingAddress: browserAddress,
+        destination: browserDestination
+      },
+      {
+        participantId: 'server',
+        signingAddress: serverAddress,
+        destination: serverDestination
+      }
+    ],
+    allocations: [
+      {
+        assetHolderAddress: ethAssetHolderAddress,
+        allocationItems: [
+          {
+            amount: formatAmount('0x5' as Uint256),
+            destination: browserDestination
+          },
+          {
+            amount: formatAmount('0x3' as Uint256),
+            destination: serverDestination
+          }
+        ]
+      }
+    ],
+    appDefinition: constants.AddressZero,
+    appData: '0x',
+    fundingStrategy: 'Direct',
+    challengeDuration: 86400 // one day
+  };
+
   await browserWallet.pushMessage(
-    serverMessageToBrowserMessage(await postFundAPromise),
+    {
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'CreateChannel',
+      params: createChannelParams
+    },
     'dummyDomain'
   );
+  const channelId = Object.keys(browserWallet.getRichObjectives())[0];
+  await browserWallet.approveRichObjective(channelId);
 
-  await objectiveSuccededPromise;
+  const serverOutput1 = await serverWallet.joinChannel({channelId});
+  await browserWallet.pushMessage(serverMessageToBrowserMessage(serverOutput1), 'dummyDomain');
 
-  await serverWallet.destroy();
-});
+  /** This is fragile. We are waiting for the third channelUpdated event. Note that these events consistently arrive in the following order.
+   *  But the events are not guaranteed to arrive in this order:
+   *  1. The first event is triggered by the registration of the channel with the chain service.
+   *      holdingUpdated is invoked when the initial holdings are read from the chain.
+   *  2. The second event is triggered by the server wallet deposit. The deposit results in a holdingUpdated invocation.
+   *  3. The third event is triggered by the browser wallet deposit.
+   */
 
-function generatePushMessage(messageParams: Message): PushMessageRequest {
-  return {
-    jsonrpc: '2.0',
-    id: 111111111,
-    method: 'PushMessage',
-    params: messageParams
-  };
-}
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const postFundA = await fromEvent<SingleChannelOutput>(serverWallet as any, 'channelUpdated')
```

Comment:
Can we can make the test less brittle by expecting something about `postFundA`?

Reply:
That would be pretty nice https://github.com/statechannels/statechannels/issues/3495

Reply:
Is there a reason not to handle it in this PR?

Reply:
... continued

it seems to me like you knew something about the value of `postFundA`, that it's currently deterministically the third event, containing some specific data. If you expect something of that specific data, then it detects regressions from that assumption.

Reply:
We can. My preference is to merge this work and then address this as a follow up.

Reply:
Your call!